File: 백준/Silver/1541. 잃어버린 괄호/잃어버린 괄호.cs
using System;

public class MainClass {
    public static void Main(string[] args) {
        string input = Console.ReadLine();
        string[] parts = input.Split('-');
        int res = int.MaxValue;
        foreach (string part in parts) {
            int temp = 0;
            string[] nums = part.Split('+');
            foreach (string num in nums) {
                temp += int.Parse(num);
            }
            if (res == int.MaxValue)
                res = temp;
            else
                res -= temp;
        }
        Console.WriteLine(res);
    }
}

File: 백준/Silver/1541. 잃어버린 괄호/잃어버린 괄호.py
import re
def solve():
    u = []
    s = input().split('-')
    for i in s:
        res = 0
        nums = i.split('+')
        for j in nums:
            res+=int(j)
        u.append(res)
    n = u[0]
    for i in range(1, len(u)):
        n -= u[i]
    return n

print(solve())
